spec/models/miq_filter_spec.rb

```diff
@@ -0,0 +1,52 @@
+describe MiqFilter do
+  let(:ems)             { FactoryGirl.create(:ems_vmware, :name => 'ems') }
+  let(:ems_folder_path) { "/belongsto/ExtManagementSystem|#{ems.name}" }
+  let(:datacenter)      { FactoryGirl.create(:ems_folder, :name => "Datacenters") }
+  let(:mtc)             { FactoryGirl.create(:ems_folder, :name => "MTC", :is_datacenter => true) }
+  let(:ems_folder_path) { "/belongsto/ExtManagementSystem|#{ems.name}" }
+  let(:mtc_folder_path) { "#{ems_folder_path}/EmsFolder|#{datacenter.name}/EmsFolder|#{mtc.name}" }
+  let(:host_folder)     { FactoryGirl.create(:ems_folder, :name => "host") }
+  let(:host_1)          { FactoryGirl.create(:host_vmware, :name => "Host_1", :ext_management_system => ems) }
+  let(:host_2)          { FactoryGirl.create(:host_vmware, :name => "Host_2", :ext_management_system => ems) }
+  let(:host_3)          { FactoryGirl.create(:host_vmware, :name => "Host_3") }
+
+  let(:mtc_folder_path_with_host_folder) { "#{mtc_folder_path}/EmsFolder|host" }
+  let(:mtc_folder_path_with_host_1)      { "#{mtc_folder_path_with_host_folder}/Host|#{host_1.name}" }
+
+  before do
+    datacenter.parent = ems
+    mtc.parent = datacenter
+    host_folder.parent = mtc
+
+    host_1.parent = host_folder
+    host_2.parent = host_folder
+  end
+
+  describe ".apply_belongsto_filters" do
+    it "returns all parent's Host objects when most-specialized filter is the Host object" do
+      input_objects = [host_1, host_2]
+      results = MiqFilter.apply_belongsto_filters(input_objects, [mtc_folder_path_with_host_1])
+      expect(results).to match_array(input_objects)
+    end
+
+    it "returns all Host objects when most-specialized filter is Host folder" do
```

Comment:
@lpichler thanks for writing these tests.  In a followup PR can you clarify the difference between these two test cases, it's not really clear why we get the same results `input_objects`... `[host_1, host_2]` but we have `returns all Host objects` vs. `returns all parent's Host objects`.